src/ServerHandler.cpp

```diff
@@ -348,17 +318,15 @@ void ServerHandler::runServers(std::map<std::string, Server*> &servers)
 		removeTimedOutClientsAndConnections();
 		// Wait max 3 minutes for incoming traffic
 		int result = poll(_pollfds.data(), _pollfds.size(), CONNECTION_TIMEOUT);
-		if (result == 0)
-		{
-			closeConnections();
-			throw TimeOutException("The program excited with timeout");
-		}
-		else if (result < 0)
+		if (result < 0)
```

Comment:
is there a case where it equals zero?

Reply:
nope, we dont want to quit even if we dont get any traffic

Reply:
result = 0 means we just didnt get any traffic in or out, so lets handler requests we already have and see then if there is more requests coming in.